internal/rows/arrowbased/batchloader_test.go

```diff
@@ -0,0 +1,89 @@
+package arrowbased
+
+import (
+	"bytes"
+	"context"
+	"github.com/apache/arrow/go/v11/arrow"
+	"github.com/apache/arrow/go/v11/arrow/array"
+	"github.com/apache/arrow/go/v11/arrow/ipc"
+	"github.com/apache/arrow/go/v11/arrow/memory"
+	"github.com/databricks/databricks-sql-go/internal/cli_service"
+	"github.com/databricks/databricks-sql-go/internal/config"
+	"github.com/pkg/errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+func generateMockArrowBytes() []byte {
+	mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer mem.AssertSize(nil, 0)
+
+	fields := []arrow.Field{
+		{Name: "id", Type: arrow.PrimitiveTypes.Int32},
+		{Name: "name", Type: arrow.BinaryTypes.String},
+	}
+	schema := arrow.NewSchema(fields, nil)
+
+	builder := array.NewRecordBuilder(mem, schema)
+	defer builder.Release()
+
+	builder.Field(0).(*array.Int32Builder).AppendValues([]int32{1, 2, 3}, nil)
+	builder.Field(1).(*array.StringBuilder).AppendValues([]string{"one", "two", "three"}, nil)
+
+	record := builder.NewRecord()
+
+	var buf bytes.Buffer
+	w := ipc.NewWriter(&buf, ipc.WithSchema(record.Schema()))
+	if err := w.Write(record); err != nil {
+		return nil
+	}
+	if err := w.Close(); err != nil {
+		return nil
+	}
+	return buf.Bytes()
+}
+
+func TestBatchLoader(t *testing.T) {
+	testTable := []struct {
+		name             string
+		server           *httptest.Server
+		expectedResponse []*sparkArrowBatch
+		expectedErr      error
+	}{
+		{
+			name: "cloud-fetch-happy-case",
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+				w.Write(generateMockArrowBytes())
+			})),
+			expectedResponse: nil, // replace with []*sparkArrowBatch{}
+			expectedErr:      nil,
+		},
+	}
+
+	for _, tc := range testTable {
+		t.Run(tc.name, func(t *testing.T) {
+			defer tc.server.Close()
+
+			cu := &cloudURL{
+				TSparkArrowResultLink: &cli_service.TSparkArrowResultLink{
+					FileLink: tc.server.URL,
+				},
+			}
+
+			ctx := context.Background()
+			cfg := config.WithDefaults()
+
+			resp, err := cu.Fetch(ctx, cfg)
```

Comment:
At this point, I think I am properly connecting to test server defined here but `resp` is always `nil` and `err` is "EOF"

Reply:
When I try to run this I'm getting `runtime error: invalid memory address or nil pointer dereference`
Adding the line in `generateMockArrowBytes` seems to fix this and the test runs successfully.  Of course the test itself fails because there's no `expectedResponse` in the test case

Reply:
Could I ask what you added in `generateMockArrowBytes` to fix that?